standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1050,17 +1073,26 @@ private List<Partition> getPartitionsFromPartitionIds(String catName, String dbN
             + ".\"SD_ID\" " + "  left outer join " + SERDES + " on " + SDS + ".\"SERDE_ID\" = "
             + SERDES + ".\"SERDE_ID\" " + "where \"PART_ID\" in (" + partIds
             + ") order by \"PART_NAME\" asc";
+    // Keep order by name, consistent with JDO.
+    ArrayList<Partition> orderedResult = new ArrayList<Partition>(partIdList.size());
+    populatePartitionsByQuery(catName, dbName, tblName, isView, queryText, null, projectionFields,
+        isAcidTable, args, orderedResult);
+    return orderedResult;
+  }
+
+  private void populatePartitionsByQuery(String catName, String dbName, String tblName,
```

Comment:
why not return `List<Partition> getPartitionsByQuery(..)` instead of passing it as an input param?

Reply:
Addressed. The initial thought was that the result list size can be determined by partition names size, so I want to init the result list before calling `populatePartitionsByQuery`, but actually the result size can be fetched from the sqlResult.